src/dialect.js

```diff
@@ -408,43 +413,42 @@ class Dialect {
    * @return Map                   A Map of escaped fields.
    */
   escapes(names, prefix, aliases) {
+    const sql = new Map()
+    let name;
     names = Array.isArray(names) ? names : [names];
-    var sql = new Map(), name, key, len, str;
-    len = names.length
 
-    for (var i = 0; i < len; i++) {
-      var value = names[i];
+    for (const value of names) {
       if (typeof value === 'string') {
         name = this.name(value, aliases);
         name = prefix ? prefix + '.' + name : name;
         sql.set(name, name);
       } else if (Array.isArray(value)) {
-        var map = this.escapes(value, prefix, aliases);
-        map.forEach(function(v, k) {
+        const map = this.escapes(value, prefix, aliases);
+        for(const [k, v] of map.entries()) {
           sql.set(k, v);
-        });
+        }
       } else if (value && value.constructor === Object) {
```

Comment:
`typeof value === 'object'`?

Reply:
I think this is trying to determine if it is a plain object or a statement instance...